service/worker/scanner/executor/executor.go

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package executor
+
+import (
+	"sync"
+	"sync/atomic"
+)
+
+type (
+	// Task defines the interface for a runnable task
+	Task interface {
+		// Run should execute the task and return well known status codes
+		Run() TaskStatus
+	}
+
+	// Executor defines the interface for any executor which can
+	// accept tasks and execute them based on some policy
+	Executor interface {
+		// Start starts the executor
+		Start()
+		// Stop stops the executor
+		Stop()
+		// Submit is a blocking call that accepts a task to execute
+		Submit(task Task) bool
+		// TaskCount returns the number of outstanding tasks in the executor
+		TaskCount() int64
+	}
+
+	// fixedPoolExecutor is an implemenation of an executor that uses fixes size
+	// goroutine pool. This executor also supports deferred execution of tasks
+	// for fairness
+	fixedPoolExecutor struct {
+		size        int
+		runQ        *runQueue
+		outstanding int64
```

Comment:
It cannot be because of the definition of what outstanding means here - outstanding is the number of tasks that are currently being executed or waiting to be executed in the executor. The queue cannot keep track of tasks which are dequeued and currently being processed

I will add a comment to clarify this